spock-specs/src/test/groovy/org/spockframework/smoke/mock/JavaMocksForGroovyClasses.groovy

```diff
@@ -88,6 +89,33 @@ class JavaMocksForGroovyClasses extends Specification {
     value2 == "value2"
   }
 
+  /**
+   This is not really the correct test, but it is related to:
+   Better way to distinguish go.x and go.getProperty("x") in JavaMockInterceptor
+   https://github.com/spockframework/spock/issues/1076
+   and
+   https://github.com/spockframework/spock/pull/1717
+   where Groovy 4.0.7 changed the compilation to indy, which makes the mockMe.getProperty("bar") call not distinguishable from mockMe.bar
+   */
+  @Requires({ GroovyRuntimeUtil.isGroovy4orNewer() })
+  def "mock call to GroovyObject.getProperty Groovy >=4.0.7"() {
+    when:
+    def value = mockMe.getProperty("bar")
+
+    then:
+    //FIXME: this should be  1 * mockMe.getProperty("bar") >> "value" but does not work in Groovy >=4.0.7
+    1 * mockMe.getBar() >> "value"
+    value == "value"
+
+    when:
+    def value2 = mockMe.getProperty("bar")
+
+    then:
+    //FIXME: this should be  1 * mockMe.getProperty("bar") >> "value" but does not work in Groovy >=4.0.7
+    1 * mockMe.getBar() >> "value2"
+    value2 == "value2"
```

Comment:
Why do we test the same thing twice? Did you intend to test this instead?
```suggestion
    def value2 = mockMe.getBar()

    then:
    1 * mockMe.getBar() >> "value2"
    value2 == "value2"
```

Reply:
Because the old test also did this twice. I just copied the Test one above to make it the Groovy 4 test.